File: RewardSystemJava/RewardSystemjava/src/main/java/provanasservices/rewardsystem/RewardMob.java
package provanasservices.rewardsystem;

import java.util.ArrayList;
import java.util.HashMap;

public class RewardMob {
    private int radius = -1;
    private ArrayList<String> rewardMessages;
    private String name;
    private String type;
    private String enabledWorld;
    private String enabledRegion;
    private ArrayList<String> allRewards;
    private HashMap<Integer, ArrayList<String>> rewards = new HashMap<>();

    public String getName() {
        return name;
    }

    public boolean nameEquals(String name) {
        if(this.name == null) return true;
        return this.name.equals(name);
    }
    public void setName(String name) {
        this.name = name;
    }

    public String getType() {
        return type;
    }

    public boolean typeEquals(String type) {
        if(this.type == null) return true;
        return this.type.equals(type);
    }
    public void setType(String type) {
        this.type = type;
    }

    public String getEnabledWorld() {
        return enabledWorld;
    }

    public boolean worldEquals(String world) {
        if(this.enabledWorld == null) return true;
        return this.enabledWorld.equals(world);
    }
    public void setEnabledWorld(String enabledWorld) {
        this.enabledWorld = enabledWorld;
    }

    public String getEnabledRegion() {
        return enabledRegion;
    }

    public boolean regionEquals(String region) {
        if(this.enabledRegion == null) return true;
        return this.enabledRegion.equals(region);
    }
    public void setEnabledRegion(String enabledRegion) {
        this.enabledRegion = enabledRegion;
    }

    public ArrayList<String> getAllRewards() {
        return allRewards;
    }
    public void setAllRewards(ArrayList<String> allRewards) {
        this.allRewards = allRewards;
    }

    public HashMap<Integer, ArrayList<String>> getRewards() {
        return rewards;
    }

    public void setRewards(HashMap<Integer, ArrayList<String>> rewards) {
        this.rewards = rewards;
    }

    public ArrayList<String> getRewardMessages() {
        return rewardMessages;
    }

    public void setRewardMessages(ArrayList<String> rewardMessages) {
        this.rewardMessages = rewardMessages;
    }

    public int getRadius() {
        return radius;
    }

    public void setRadius(int radius) {
        this.radius = radius;
    }
}

File: app/src/main/kotlin/provanasservices/rewardsystem/RewardMob.kt
package provanasservices.rewardsystem

import java.util.*
import kotlin.collections.ArrayList
import kotlin.collections.HashMap
import net.md_5.bungee.api.ChatColor

class RewardMob {
    var id: String = ""
    var radius = -1
    var rewardMessages: List<String>? = null
    var name: String? = null
    var type: String? = null
    var enabledWorld: String? = null
    var enabledRegion: String? = null
    var allRewards: List<String>? = null
    var rewards = HashMap<Int, List<String>>()
    val chanceRewards = HashMap<Int, List<ChanceReward>>()
    var allChanceRewards = ArrayList<ChanceReward>()
    var minimumDamage = 0.0
    var cooldown = 0
    var cooldowns: MutableMap<UUID, Long> = Collections.synchronizedMap(HashMap())
    var cooldownMessage = ""
    fun nameEquals(name: String): Boolean {
        return if (this.name == null) true else {
            val strippedInputName = ChatColor.stripColor(ChatColor.translateAlternateColorCodes('&', name))
            val strippedConfigName = ChatColor.stripColor(ChatColor.translateAlternateColorCodes('&', this.name!!))

            val plugin = Main.getInstance()
            if (plugin.config.getBoolean("Debug.enabled")) {
                plugin.logger.info("[REWARDSYSTEM DEBUG] Name comparison - Input: '$name', Config: '${this.name}'")
                plugin.logger.info("[REWARDSYSTEM DEBUG] Stripped - Input: '$strippedInputName', Config: '$strippedConfigName'")
            }

            this.name == name || strippedConfigName == strippedInputName
        }
    }

    fun typeEquals(type: String): Boolean {
        return if (this.type == null) true else this.type == type
    }

    fun worldEquals(world: String): Boolean {
        return if (enabledWorld == null) true else enabledWorld == world
    }

    data class ChanceReward(val chance: Double?, val commands: String, val chancePlaceholder: String? = null)
}